cli/cli/conf.py

```diff
@@ -1,4 +1,82 @@
 import os
+import typing as t
+from dataclasses import asdict, dataclass
+
+import yaml
+
+if t.TYPE_CHECKING:
+    # Importing conditionally to avoid circular imports
+    from cli.infra import InfraManager
 
 USER_HOME = os.path.expanduser("~")
 CONFIG_FILE = os.path.join(USER_HOME, ".config", "scw", "gateway.yml")
+
+# Name is fixed for Scaleway managed database
+DB_DATABASE_NAME = "rdb"
+DB_DATABASE_NAME_LOCAL = "kong"
+
+
+@dataclass
+class InfraConfiguration:
+    """Configuration used to store information of a deployed gateway."""
+
+    protocol: t.Literal["http", "https"]
+    gw_admin_host: str
+    gw_admin_port: str
+    gw_admin_token: str
+    gw_host: str
+    gw_port: str
+    db_host: str
+    db_port: str
+    db_name: str
+
+    @staticmethod
+    def from_local() -> "InfraConfiguration":
+        """Get the configuration for the local docker-compose stack."""
+        return InfraConfiguration(
+            protocol="http",
+            gw_admin_host="localhost",
+            gw_admin_port="8001",
+            gw_admin_token="",
+            gw_host="localhost",
+            gw_port="8080",
+            db_host="localhost",
+            db_port="5432",
+            db_name=DB_DATABASE_NAME_LOCAL,
+        )
+
+    @staticmethod
+    def from_infra(manager: "InfraManager") -> "InfraConfiguration":
```

Comment:
To avoid circular imports, we can move this to a method on the `InfraManager` class called something like `build_config`.